File: eRecipes/UI/erecipes_mobile/lib/models/recept_sastojak.dart
import 'package:erecipes_mobile/models/recept.dart';
import 'package:erecipes_mobile/models/sastojak.dart';
import 'package:json_annotation/json_annotation.dart';

part 'recept_sastojak.g.dart';

@JsonSerializable()
class ReceptSastojak{
  int? receptSastojakId;
  int? receptId; 
 int? sastojakId;
Recept? recept;
Sastojak? sastojak; 
  ReceptSastojak(
    {
          required this.receptId,
          required this.sastojakId,
           required this.sastojak
    });

 factory ReceptSastojak.FromJson(Map<String,dynamic> json)=> _$ReceptSastojakFromJson(json);

 Map<String,dynamic> toJson() => _$ReceptSastojakToJson(this);
 Sastojak toSastojak() {
    return this.sastojak!; // Pretpostavljamo da je sastojak sigurno prisutan
  }
}

File: eRecipes/eRecipes.Model/ReceptSastojak.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eRecipes.Model
{
    public class ReceptSastojak
    {
        public int ReceptSastojakId { get; set; }
        public int ReceptId { get; set; }
        public int SastojakId { get; set; }
        public virtual Recept? Recept { get; set; }
        public virtual Sastojak? Sastojak { get; set; }
        public int MjernaJedinicaId { get; set; }
        public double Kolicina { get; set; }
        public virtual MjernaJedinica MjernaJedinica { get; set; } = null!;
    }
}
